tests/client/core/components/TestPaginatorLink.js

```diff
@@ -0,0 +1,121 @@
+
+import PaginatorLink from 'core/components/PaginatorLink';
+
+describe('<PaginatorLink />', () => {
+  const renderProps = Object.freeze({
+    currentPage: 2,
+    page: 3,
+    pageCount: 4,
+    pathname: '/some/link',
+  });
+
+  function renderLink(customProps = {}) {
+    const props = { ...renderProps, ...customProps };
+    return findDOMNode(
+      renderIntoDocument(<PaginatorLink {...props} />));
+  }
+
+  it('requires currentPage', () => {
+    const props = { ...renderProps };
+    delete props.currentPage;
+    assert.throws(
+      () => renderIntoDocument(<PaginatorLink {...props} />),
+      /currentPage .* cannot be undefined/);
+  });
+
+  it('requires pathname', () => {
+    const props = { ...renderProps };
+    delete props.pathname;
+    assert.throws(
+      () => renderIntoDocument(<PaginatorLink {...props} />),
+      /pathname .* cannot be undefined/);
+  });
+
+  it('requires a page', () => {
+    const props = { ...renderProps };
+    delete props.page;
+    assert.throws(
+      () => renderIntoDocument(<PaginatorLink {...props} />),
+      /page .* cannot be undefined/);
+  });
+
+  it('requires pageCount', () => {
+    const props = { ...renderProps };
+    delete props.pageCount;
+    assert.throws(
+      () => renderIntoDocument(<PaginatorLink {...props} />),
+      /pageCount .* cannot be undefined/);
+  });
+
+  describe('when the link is to the current page', () => {
+    it('does not contain a link', () => {
+      const item = renderLink({ currentPage: 3, page: 3 });
+      assert.strictEqual(item.querySelector('a'), null);
+      assert.equal(item.textContent, '3');
+    });
+
+    it('uses the provided text', () => {
+      const item = renderLink({
+        currentPage: 3, page: 3, text: 'go to page',
+      });
+      assert.equal(item.tagName, 'SPAN');
+      assert.equal(item.textContent, 'go to page');
+    });
+  });
+
+  describe('when the link is to a different page', () => {
+    it('has a link', () => {
+      const link = renderLink({ page: 3 });
+      assert.equal(link.tagName, 'A');
+      assert.equal(link.textContent, '3');
+    });
+
+    it('uses the provided text', () => {
+      const link = renderLink({ page: 3, text: 'go to next page' });
+      assert.equal(link.tagName, 'A');
+      assert.equal(link.textContent, 'go to next page');
+    });
+
+    it('renders a custom link component', () => {
+      const LinkComponent = sinon.spy(() => <div />);
+      renderLink({ page: 3, LinkComponent });
+      assert.ok(LinkComponent.called, 'custom LinkComponent was not rendered');
+    });
+
+    it('passes query params to the link', () => {
+      const page = 3;
+      const queryParams = { color: 'red' };
+      const LinkComponent = sinon.spy(() => <div />);
+      renderLink({ queryParams, page, LinkComponent });
+
+      assert.ok(LinkComponent.called, 'custom LinkComponent was not rendered');
+      const props = LinkComponent.firstCall.args[0];
```

Comment:
This is a nitpick but I kinda like when simpler tests like this declare things at the top, then do all their assertions at the bottom in succession.

It makes it easier to scan, but also doesn't imply anything is changing between the assertions. Because here nothing is different between `assert.ok(LinkComponent.called)` and `assert.deepEqual(props.to.query, [...])`, I think moving the `const props = [...]` line above the assertions makes sense.

We have more complex tests where we test state before/after a click, and that's when I expect to see variable definitions, etc. between assertions.